streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -17,378 +17,831 @@
-     * will use the generated name of the parent processor node to label the key/value pairs
-     * printed out to the console.
-     *
-     * Implementors will need to override toString for keys and values that are not of
-     * type String, Integer etc to get meaningful information.
+     * Print the update records of this {@link KTable} to {@code System.out}.
+     * This function will use the generated name of the parent processor node to label the key/value pairs printed out
+     * to the console.
+     * <p>
+     * The provided serde will be used to deserialize the key or value in case the type is {@code byte[]} before calling
+     * {@code toString()} on the deserialized object.
+     * <p>
+     * Implementors will need to override {@code toString()} for keys and values that are not of type {@link String},
+     * {@link Integer} etc. to get meaningful information.
+     * <p>
+     * Note that {@code print()} is not applied to the internal state store and only called for each new {@link KTable}
+     * update record.
      */
     void print();
 
     /**
-     * Print the elements of this stream to {@code System.out}.  This function
-     * will use the given name to label the key/value printed out to the console.
+     * Print the update records of this {@link KTable} to {@code System.out}.
+     * This function will use the given name to label the key/value pairs printed out to the console.
+     * <p>
+     * The provided serde will be used to deserialize the key or value in case the type is {@code byte[]} before calling
+     * {@code toString()} on the deserialized object.
+     * <p>
+     * Implementors will need to override {@code toString()} for keys and values that are not of type {@link String},
+     * {@link Integer} etc. to get meaningful information.
+     * <p>
+     * Note that {@code print()} is not applied to the internal state store and only called for each new {@link KTable}
+     * update record.
      *
      * @param streamName the name used to label the key/value pairs printed out to the console
-     *
-     * Implementors will need to override toString for keys and values that are not of
-     * type String, Integer etc to get meaningful information.
      */
-    void print(String streamName);
+    void print(final String streamName);
 
     /**
-     * Print the elements of this stream to {@code System.out}
-     * @param keySerde key serde used to send key-value pairs,
-     *                 if not specified the default serde defined in the configs will be used
-     * @param valSerde value serde used to send key-value pairs,
-     *                 if not specified the default serde defined in the configs will be used
-     *
-     * Implementors will need to override toString for keys and values that are not of
-     * type String, Integer etc to get meaningful information.
+     * Print the update records of this {@link KTable} to {@code System.out}.
```

Comment:
Was intended as is -- records are updates, so its an "update record" (or update-record). "updated" makes sense, too, though. Not sure what the better wording is...